packages/platform-browser/animations/async/src/async_animation_renderer.ts

```diff
@@ -191,6 +217,11 @@ export class DynamicDelegationRenderer implements Renderer2 {
   }
 
   listen(target: any, eventName: string, callback: (event: any) => boolean | void): () => void {
+    // We need to keep track of animation events registred by the default renderer
+    // So we can also register them against the animation renderer
+    if (this.animationEvents !== null && eventName.startsWith('@')) {
+      this.animationEvents.push({target, eventName, callback});
+    }
```

Comment:
Just as a refactoring idea: we can probably have a single array where we collect functions that should be invoked later once "real" renderer is loaded, for example:

```suggestion
    if (this.shouldReplay(eventName)) {
      this.replay.push((renderer: Renderer2) => renderer.listen(target, eventName, callback));
    }
```
The `shouldReplay` function may look like this:
```
private shouldReplay(propOrEventName: string): boolean {
  return this.replay !== null && propOrEventName.startsWith(ANIMATION_PREFIX);
}
```
After that in the `use` function we can do something like this:
```
  use(impl: BaseAnimationRenderer) {
    this.delegate = impl;

    if (this.replay !== null) {
      // Replay queued actions using loaded renderer instance to apply
      // all events and properties collected while loading was in progress.
      for (const fn of this.replay) {
        fn(impl);
      }
      // Set to `null` to indicate that the queue was processed
      // and we no longer need to collect events and properties.
      this.replay = null;
    }
  }
```

Reply:
A couple additional comments (applicable to the current implementation too):
- we should cleanup the queue in the `destroy` method
- we'd need to account for a situation when loading of the real renderer fails, in which case we should stop collecting callbacks and clear the queue, so that we don't have memory leaks (+ we'd need to output `console.error` and mention that loading failed and no animations would be displayed)

Reply:
Great suggestions, 👍